src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureKeyVaultReference/AzureKeyVaultSecretProvider.cs

```diff
@@ -33,13 +36,8 @@ public AzureKeyVaultSecretProvider(TokenCredential credential = null, IEnumerabl
             }
         }
 
-        public async Task<string> GetSecretValue(Uri secretUri, CancellationToken cancellationToken)
+        public async Task<string> GetSecretValue(Uri secretUri, string key, CancellationToken cancellationToken)
```

Comment:
>string key [](start = 64, length = 10)

So if we rotate the underlying secret for a given key, other keys that use the same underlying secret won't get updated if they don't specify `SetSecretRefreshInterval`? It might make sense, but I expected the secret to get rotated for any keys that used it. It may be worth seeing what others expected.

Reply:
Not sure why someone will reference the same secret with two different keys, but yes, that behavior will be as currently designed. Hard to say which behavior will be expected since the user specifically tells us to refresh one but not the other. If we have a scenario in mind, maybe it's easier to decide.

QQ, if people do set refresh interval on both keys, will we refresh the same secret twice?

Reply:
Yes, we will refresh the same secret twice if both keys point to the same Key Vault reference. If a refresh interval is set, we will honor it and refresh the secret for every unique key. We don't check what the Key Vault reference is while refreshing secrets.

Reply:
I think that's fine. I just wanted to make sure we are on the same page.

Reply:
The problem can be solved if we introduce references to other keys in appconfig in the future.